### PR TITLE
Fix possible access violation in storage builder

### PR DESCRIPTION
**Pull request by anstaf**

### CODE REVIEW

**Comment by anstaf** [comment]
launch jenkins

File: include/gridtools/storage/builder.hpp [hunk 1]
```diff
@@ -38,10 +38,15 @@ namespace gridtools {
             void initializer_impl(
                 Fun const &fun, T *dst, Layout layout, info<N> const &info, std::index_sequence<Is...>) {
                 int length = info.length();
+                auto &&lengths = info.lengths();
 #pragma omp parallel for
                 for (int i = 0; i < length; ++i) {
                     auto indices = info.indices(layout, i);
+                    for (auto ok : {(indices[Is] < lengths[Is])...})
```
Comment (havogt) [inline include/gridtools/storage/builder.hpp:45]: I thought I commented already yesterday, but apparently didn't commit.
I appreciate that I learned a new pattern (range-for-loop over initializer list created from pack expansion)!
However, I'd prefer a more expressive way for sake of maintainability by outlining to a named function/lambda, such that it reads something like
```
if(in_range(...))
   dst[i] = fun(indices[Is]...);
```

Comment (anstaf) [reply include/gridtools/storage/builder.hpp]: done

**Comment by anstaf** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
I'll merge. (failed because tsa not available)

**Pull request merged**